### DISCUSSION TITLE
Use local dns in "forwarding-rules"

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [DNSCrypt/dnscrypt-proxy#1964] [Potential issues]
I have specific domains that would like to resolve use locally configured DNS, instead of going through dnscrypt-proxy.
How should I configure it in "forwarding-rules"?

**Comment by ianbashford** [comment]
Hi @CaledoniaProject 

The examples in the file: [`example-forwarding-rules.txt`](https://github.com/DNSCrypt/dnscrypt-proxy/blob/master/dnscrypt-proxy/example-forwarding-rules.txt)   are hopefully pretty clear - each line has the format
`<domain> <server address>[:port]`

My hosts are assigned to the domain '.local' , so I use the line

```
local          192.168.0.1
```
The IP address needs to be the address of your router or your locally configured DNS - they'll receive the cleartext DNS Query.

**Comment by CaledoniaProject** [comment]
I meant to use whatever DNS server returned from DHCP, instead of going through DoH.
I need to use the intranet DNS servers when I'm in the office, and another preconfigured DNS server when I'm at home.

Preferred config is like:

```
google.com auto;
```

**Comment by ianbashford** [comment]
If you're trying to avoid passing local domains through dnscrypt-proxy, then the forwarding rules won't help -- that's a convenience that allows you to configure as many specific resolvers by domain as as you might need and to use an external resolver in all other cases.  It's a powerful feature and it has the advantage of working across any platform - windows/macos/linux etc.

It sounds like you're trying to address this at the operating system level instead - there's something on the link below about REHL being able to do that -- I assume any systemd based linux could do the same.
Someone else on here may know how to do this on other systems... 

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/using-different-dns-servers-for-different-domains_configuring-and-managing-networking
